src/js/preshuffle.s

```diff
@@ -1398,14 +1415,20 @@ UpdateEnemyHP:
     ; Check to see if the monster we attacked is on the blocklist
     ; by looking to see if it has a name. If the name table hi addr is $00
     ; then we don't want to display it (even if it does have HP)
-    ldx ObjectNameId,y
-    lda EnemyNameTableHi,x
-    beq +
-      sty CurrentEnemySlot
-      lda ShouldRedisplayUI
-      ora #DRAW_ENEMY_STATS
-      sta ShouldRedisplayUI
-+   rts
+    lda ObjectNameId,y
+    ; The blocklist should never be empty, but if it somehow does, then this code would break horribly
+.assert ENEMY_NAME_BLOCKLIST_LEN > 0
+    ldx #ENEMY_NAME_BLOCKLIST_LEN-1
+-     cmp EnemyNameBlocklist,x
+      beq @Exit
+      dex
+      bpl -
+    sty CurrentEnemySlot
+    lda ShouldRedisplayUI
+    ora #DRAW_ENEMY_STATS
+    sta ShouldRedisplayUI
+@Exit:
```

Comment:
Without any real labels in between, this is ambiguous with the earlier `@Exit:` above.  It also confused me for a bit while the other version was folded away.  Can we give them more unique names or else put a permanent label in between?

This is still a small enough jump that a positional `+` label (along with indentation) would be completely clear.